docs/HowToDebugCrashWithAddr2lineTool.md

```diff
@@ -0,0 +1,116 @@
+Simple command is like below.  
+```
+$ addr2line -e [filepath] -a [address]
+```
+
+However, if the path to your toolchain is not configured as mentioned above, or if there are multiple toolchains installed on your system,
+then you might have to speficy the correct version of the addr2line to be used. It can be done as follows:
+
+Below command specifies that ARM toolchain is to be used.
+For this to work, the ARM toolchain needs to be configured in the PATH:
+```
+$ arm-none-eabi-addr2line -e [filepath] -a [address]
+```
+
+However, if the toolchain path is not configured, or if you have mulitple versions of toolchain, then you may speficy the complete path to the
+required addr2line tool as follows:
+```
+$ /home/tools/gcc-arm-none-eabi-9-2019-q4-major/bin/arm-none-eabi-addr2line -e [filepath] -a [address]
+```
+
+### How to use in Binary Separation
+When binary separation, the address of text section in user binary starts from 0x0 for relocation.  
+After relocation, its real address is allocated and those address can be used for debugging.  
+For this, some steps are required like below.  
+
+1. Enable these configs:
+
+	CONFIG_SAVE_BIN_SECTION_ADDR  
+	```
+	menuconfig -> Binary Loader -> Save binary section address
+	```
+
+	CONFIG_DEBUG_BINFMT_INFO
+	```
+	menuconfig -> Debug Options -> Enable Debug Features -> Binary Loader Debug Features -> Binary Loader Informational Output
+	```
+
+2. Check the user binary's text address from the printed log.  
+
+3. Calculate the address based on No.2.  
+For example, let assume that user binary text address is 0x1000000, and symbol address ix 0x1200000.  
+Then subtract 0x1000000 from 0x1200000 and pass this address for -a option's parameter.
+	```
+	$ addr2line -e [filepath] -a 0x200000
+	```
+
+## How to debug a crash with addr2line
+When a crash occurs in the system, it is useful to find out where exactly the crash occured and also
+the function callstack which resulted in the crash. In order to find these, we need to print the corresponding
+address values when the crash occurs. This can be done by enabling the following configs:
+
+CONFIG_ARCH_STACKDUMP
+```
+menuconfig -> Chip selection -> Dump stack on assertions
+```
+
+When a crash occurs in the system, and if the above configs are enabled, then the system prints the stackdump for the currently
+running task in which the crash occured and also the register dump showing the contents of the CPU registers at the time of crash.
+You can then use the addr2line command as stated above to convert the printed address values into source file line numbers and the
+following debug information can be obtained.
+
+1. Location where the crash occured
+This can be found from the register dump. The register dump contains the values of R0 to R15 registers. The R15 register gives the
+PC (Program Counter) value. This address can be used with the addr2line command to find the exact line number in the source code
+which caused the crash.
```

Comment:
Will adding a sample crash log (or just register values from R0-R15 pointing to R15 value) be useful ?
What is your opinion ?

Reply:
Done